File: app/src/main/java/net/mbonnin/arcanetracker/parser/power/BlockTag.java
package net.mbonnin.arcanetracker.parser.power;

import java.util.ArrayList;
import java.util.List;

public class BlockTag extends Tag {
    public static final String TYPE_PLAY = "PLAY";
    public static final String TYPE_POWER = "POWER";
    public static final String TYPE_TRIGGER = "TRIGGER";
    public static final String TYPE_ATTACK = "ATTACK";
    public String BlockType;
    public String Entity;
    public String EffectCardId;
    public String EffectIndex;
    public String Target;
    public List<Tag> children = new ArrayList<>();
}

File: app/src/main/java/net/mbonnin/arcanetracker/parser/power/BlockTag.kt
package net.mbonnin.arcanetracker.parser.power

import java.util.ArrayList

class BlockTag(
        var BlockType: String,
        var Entity: String?,
        var EffectCardId: String,
        var EffectIndex: String,
        var Target: String?,
        var children: MutableList<Tag>,
        var SubOption: String,
        var TriggerKeyword: String?
) : Tag() {
    companion object {
        val TYPE_PLAY = "PLAY"
        val TYPE_POWER = "POWER"
        val TYPE_TRIGGER = "TRIGGER"
        val TYPE_ATTACK = "ATTACK"
    }
}
